synapse/handlers/federation.py

```diff
@@ -1498,6 +1543,42 @@ async def _sync_partial_state_room(
                 allow_rejected=True,
             )
             for event in events:
-                await self._federation_event_handler.update_state_for_partial_state_event(
-                    destination, event
-                )
+                for attempt in range(num_destinations):
+                    try:
+                        await self._federation_event_handler.update_state_for_partial_state_event(
+                            destination, event
+                        )
+                        break
+                    except (
+                        InvalidResponseError,
+                        FederationDeniedError,
+                        HttpResponseException,
+                        NotRetryingDestination,
+                        RequestSendFailed,
```

Comment:
`update_state_for_partial_state_event` raises none of these because `_resolve_state_at_missing_prevs` catches them all and raises a `FederationError` instead.